crates/uv/src/commands/project/run.rs

```diff
@@ -292,6 +310,14 @@ pub(crate) async fn run(
             return false;
         }
 
+        // `SitePackages::satisfies` doesn't handle overrides yet, and constraints are only enforced
+        // on `--with` requirements instead of the project requirements, so we perform a full
+        // resolution to ensure things are up-to-date.
+        // TODO(charlie): Support constraints and overrides in `uv run`.
+        if !(spec.constraints.is_empty() && spec.overrides.is_empty()) {
+            return false;
+        }
+
```

Comment:
This case can't occur now, right? Still worth including?

Reply:
I figured it was worth it with a TODO. I can clarify more in the comments.

Reply:
Removed